python/opentrons_ot3_firmware/messages/payloads.py

```diff
@@ -183,6 +186,21 @@ def __post_init__(self) -> None:
                 f" {FirmwareUpdateDataField.NUM_BYTES} bytes."
             )
 
+    @classmethod
+    def create(cls, address: int, data: bytes) -> "FirmwareUpdateData":
+        """Create a firmware update data payload."""
+        checksum = 0
+        obj = FirmwareUpdateData(
+            address=utils.UInt32Field(address),
+            num_bytes=utils.UInt8Field(len(data)),
+            reserved=utils.UInt8Field(0),
+            data=FirmwareUpdateDataField(data),
+            checksum=utils.UInt16Field(checksum),
+        )
+        checksum = (1 + ~sum(obj.serialize())) & 0xFFFF
```

Comment:
Why do you bitwise and this?

Reply:
it's a little clearer if you're used to reading these, if i don't see it i reflexively go "wait what happens on overflow"

Reply:
Also would `(0x0001  + ~sum(obj.serialize()))` be equivalent?

Reply:
It's a common way to perform a checksum. Add the bytes, NOT them and add 1 (make it into a negative number).

`& 0xFFFF` is to restrict the result to 16 bits.